strapi/strapi
# feat(proxy): allow the use of a proxy in internal requests

### What does it do?

Adding a new fetch function in the Strapi class.
Adding the possibility to configure a proxy through the `server.js` configuration.

```javascript
// path: ./config/server.js

module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  app: {
    keys: env.array('APP_KEYS'),
  },
  globalProxy: 'http://proxyhost:3128',
});
```

### Why is it needed?

Some users are facing issues since we moved our license check to a dedicated server.
This prevents users to run appropriately their Strapi project.

### How to test it?

1. Setup a proxy (squid for example)
2. Add the URL to this proxy in your server configuration
3. Run the app
4. You should see logs in your proxy that indicates Strapi instance doing requests

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
